pkg/cluster/internal/create/actions/installcni/cni.go

```diff
@@ -83,6 +85,40 @@ func (a *action) Execute(ctx *actions.ActionContext) error {
 		manifest = out.String()
 	}
 
+	// NOTE: this is intentionally undocumented, as an internal implementation
+	// detail. Going forward users should disable the default CNI and install
+	// their own, or use the default. The internal templating mechanism is
+	// not intended for external usage and is unstable.
+	if strings.Contains(manifest, "would you kindly patch this file") {
+		// Add the controlplane endpoint so kindnet doesn´t have to wait for kube-proxy
+		controlPlaneEndpoint, err := ctx.Provider.GetAPIServerInternalEndpoint(ctx.Config.Name)
+		if err != nil {
+			return err
+		}
+
+		patchValue := `
+- op: add
+  path: /spec/template/spec/containers/0/env/-
+  value:
+    name: CONTROL_PLANE_ENDPOINT
+    value: ` + controlPlaneEndpoint
+
+		controlPlanePatch6902 := config.PatchJSON6902{
+			Group:   "apps",
+			Version: "v1",
+			Kind:    "DaemonSet",
+			Patch:   patchValue,
+		}
+
+		patchedConfig, err := patch.KubeYAML(manifest, nil, []config.PatchJSON6902{controlPlanePatch6902})
+		if err != nil {
+			return err
+		}
+		manifest = patchedConfig
+	}
+
+	ctx.Logger.V(5).Infof("Using the following Kindnetd config on node %s:\n%s", node.String(), manifest)
```

Comment:
the node is pretty irrelevant? we're installing a daemonset.

Reply:
heh, what I was thinking